packages/transformers/js/src/visitors/dependencies.js

```diff
@@ -79,21 +87,44 @@ export default {
     }
   },
 
-  NewExpression(node, asset) {
-    const {callee, arguments: args} = node;
+  NewExpression(node, asset, ancestors) {
+    let {callee, arguments: args} = node;
 
-    const isWebWorker =
+    let isWebWorker =
       callee.type === 'Identifier' &&
       (callee.name === 'Worker' || callee.name === 'SharedWorker') &&
-      args.length === 1 &&
-      types.isStringLiteral(args[0]);
+      !hasBinding(ancestors, callee.name) &&
+      !isInFalsyBranch(ancestors) &&
+      types.isStringLiteral(args[0]) &&
+      (args.length === 1 || args.length === 2);
 
     if (isWebWorker) {
-      addURLDependency(asset, args[0], {env: {context: 'web-worker'}});
+      let isModule = false;
+      if (types.isObjectExpression(args[1])) {
+        let prop = args[1].properties.find(v =>
+          types.isIdentifier(v.key, {name: 'type'})
+        );
+        if (prop && types.isStringLiteral(prop.value))
+          isModule = prop.value.value === 'module';
+      }
+
+      addURLDependency(asset, args[0], {
+        env: {
+          context: 'web-worker',
+          outputFormat: isModule ? 'esmodule' : undefined
```

Comment:
Probably should ensure that scope-hoisting is enabled here. The packager will throw if you try to output es modules without scope hoisting enabled. The HTML transformer checks:

https://github.com/parcel-bundler/parcel/blob/14043b250a6c63cc0a80960eb0b97821c1539bf5/packages/transformers/html/src/dependencies.js#L76

Reply:
Done